src/main/java/fi/yle/tools/aws/maven/AuthenticationInfoAWSCredentialsProviderChain.java

```diff
@@ -16,20 +16,17 @@
 
 package fi.yle.tools.aws.maven;
 
-import com.amazonaws.auth.AWSCredentialsProviderChain;
-import com.amazonaws.auth.EnvironmentVariableCredentialsProvider;
-import com.amazonaws.auth.InstanceProfileCredentialsProvider;
-import com.amazonaws.auth.SystemPropertiesCredentialsProvider;
+import com.amazonaws.auth.*;
 import com.amazonaws.auth.profile.ProfileCredentialsProvider;
 import org.apache.maven.wagon.authentication.AuthenticationInfo;
 
 final class AuthenticationInfoAWSCredentialsProviderChain extends AWSCredentialsProviderChain {
 
-    AuthenticationInfoAWSCredentialsProviderChain(AuthenticationInfo authenticationInfo) {
+    AuthenticationInfoAWSCredentialsProviderChain(String awsProfile, AuthenticationInfo authenticationInfo) {
         super(new EnvironmentVariableCredentialsProvider(),
                 new SystemPropertiesCredentialsProvider(),
-                new InstanceProfileCredentialsProvider(),
-                new ProfileCredentialsProvider(),
+                new ProfileCredentialsProvider(awsProfile),
+                new EC2ContainerCredentialsProviderWrapper(),
```

Comment:
Suggest you look at doing something like this - https://github.com/verygoodsecurity/aws-maven/commit/7672449ddac96025159a9b1b93df4f027604105a#diff-bc25e29340c2c636b9eac497e8efc484R28

`DefaultAWSCredentialsProviderChain` - https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html

> AWS credentials provider chain that looks for credentials in this order:
> - Environment Variables - AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY (RECOMMENDED since they are recognized by all the AWS SDKs and CLI except for .NET), or AWS_ACCESS_KEY and AWS_SECRET_KEY (only recognized by Java SDK)
> - Java System Properties - aws.accessKeyId and aws.secretKey
> - Credential profiles file at the default location (~/.aws/credentials) shared by all AWS SDKs and the AWS CLI
> - Credentials delivered through the Amazon EC2 container service if AWS_CONTAINER_CREDENTIALS_RELATIVE_URI" environment variable is set and security manager has permission to access the variable,
> - Instance profile credentials delivered through the Amazon EC2 metadata service

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/index.html?com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html

Reply:
I know about the `DefaultAWSCredentialsProviderChain`. I would have used it, but I need to be able to overwrite the profileName, and the `DefaultAWSCredentialsProviderChain` does not allow that.

You see, in our cause we have multiple aws accounts, and our application itself needs to run under one profile, and the build under another (because our s3 bucket is located in another account).